### PR TITLE
Session

### PR DESCRIPTION
**Pull request by grant350**
session is 90% done needs to match with auth user cookie if user cookie exists or enter code new sessionCookie is created. 
AuthCookie.username-> restraunt.js   -> authcookie present -> create new token with same info in db.session;
AuthCookie = null || invalid -> sessioncode present -> look up db sessions with same code || id -> create new token cookie;

authcookie and session code is not present -> find if user is connected to session. if exists create new cookie with same session code else create new session with new session code

### CODE REVIEW

**Review by neenachcha — COMMENTED** [review]
Great job Grant,  One question from me: 
Currently in the the database. 
session_code = 1. --> should this be uuid?
orderId= "00301" --> should this be codeGenerator?

Me and yufang have been using session_code to retrieve data from database. Not sure about Tan, but would be possible to switch them

File: client/src/components/Menu/Menu.js [hunk 1]
```diff
@@ -2,16 +2,14 @@ import React from 'react';
...
 
-class Menu extends React.Component {
-	constructor(props) {
-		super(props);
-		this.state = {
-		}
-	}
-
-	render() {
+	console.log(props);
```
Comment (yfcheng1) [inline client/src/components/Menu/Menu.js:9]: Consider remove console.log

File: client/src/components/Navbar.js [hunk 2]
```diff
@@ -1,39 +1,82 @@
...
           >
             <MenuIcon />
           </IconButton>
+
           <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
-            Splitsy
+           <Link to="/" className="bob"  style={{color:"white",textDecoration:"none"}}> Splitsy</Link>
           </Typography>
-          <Button color="inherit">Login/User</Button>
+          {this.isAuthenticated? <Link className="bob"  to="/userProfile" style={{color:"white",textDecoration:"none"}}> {this.user} </Link>: null}
```
Comment (yfcheng1) [inline client/src/components/Navbar.js:75]: Can you add Typography for the {this.user}?  It looks inconsistent.

File: client/src/components/Menu/Menu.js [hunk 3]
```diff
@@ -2,16 +2,14 @@ import React from 'react';
...
+	const {state} = useLocation();
+	console.log(window.getCookie('orderSession'))
+console.log(state)
```
Comment (yfcheng1) [inline client/src/components/Menu/Menu.js:12]: Can you remove this console.log?

File: client/src/components/Restaurant/Restaurant.js [hunk 4]
```diff
@@ -1,20 +1,8 @@
...
-    )
-  }
-}
 
-export {RestaurantMenu,RestaurantPick};
\ No newline at end of file
+
+
+
+export {RestaurantPick};
```
Comment (yfcheng1) [inline client/src/components/Restaurant/Restaurant.js:8]: Do you still need this file now?

File: client/src/components/Restaurant/rest_pick/map.js [hunk 5]
```diff
@@ -34,53 +25,31 @@ class Map extends React.Component {
...
-      <span>{this.state.error}</span>
-      <div className="search-ct"><TextField error={this.state.error? this.state.error: false}  helperText={this.state.helperText} style={{margin:"10px"}} onChange={(e)=>{this.querychange(e)}} id="outlined-basic" label="" variant="outlined" value={this.state.query} className="mapsearch" />
-      <Button onClick={(e)=>{this.search(e)}} className="submit"> submit</Button>
+      <span>{this.props.query } <br/> </span>
+      <span>{this.props.error}</span>
+      <div className="search-ct">
+
+        <TextField  error={this.props.error? this.props.error: false}  helperText={this.props.helperText} style={{margin:"10px"}} onChange={(e)=>{this.props.querychange(e,"query")}} id="outlined-basic" label="address" variant="outlined" value={this.props.query} className="mapsearch" />
+        <TextField error={this.props.error? this.props.error: false}  helperText={this.props.helperText} style={{margin:"10px"}} onChange={(e)=>{this.props.querychange(e,"keywords")}} id="outlined-basic" label="keywords" variant="outlined" value={this.props.keyword} className="mapsearch" />
+        OR
```
Comment (yfcheng1) [inline client/src/components/Restaurant/rest_pick/map.js:44]: Can you use typescript here?

File: client/src/components/Restaurant/rest_pick/list.js [hunk 6]
```diff
@@ -4,18 +4,60 @@
...
+        }
+        props.clickRestaurant(obj, ()=>{
+          navigate('/menu', { state: { item:item } });
+        })
+
+      }
+      var retryindex=1;
+      //state={{ menu: item }}
+      return  ( <div onClick={clicked}>
+       <ListItem key={index} className="item" button style={{height:"150px", border:"1px solid black"}}>
```
Comment (yfcheng1) [inline client/src/components/Restaurant/rest_pick/list.js:32]: Can you remove the border? or make it grey to match the style

File: client/src/components/Restaurant/rest_pick/rest_pick.js [hunk 7]
```diff
@@ -1,22 +1,122 @@
...
+      // this.setState({rest:data.data})
+    });
+  }
+
+
+
   render() {
     return (
       <div className="rest">
       <p> Restaurants near me </p>
```
Comment (yfcheng1) [inline client/src/components/Restaurant/rest_pick/rest_pick.js:116]: Can you use typography here?

File: server/controllers/orderSession/index.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+      return symbols;
+    }
+    var session_code = codeGenerator();
+
+    // make requests to find userid in db
+    var obj = {
+      session_code:session_code,
+      restaurant: {'restaurant_id': index, 'searchNear':address, 'address':req.query.street_address, 'name': req.query.name},
+      orderId:uuidv4(),
+      users:{username:username,checkout:false,user_cart:[]},
```
Comment (yfcheng1) [inline server/controllers/orderSession/index.js:37]: users should be nested objects. Something like:
users: { 
  user_id: {
    username:username,
    checkout:false,
    user_cart:[]
    }
}

File: server/controllers/orderSession/index.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+      // // //for sessionuuid code
+      // var token = jwt.sign(payload,"server password",{ expiresIn: '1h' });
+      // res.send(token)
+    })
+    // db.sessionSchema.create(obj).then(result=>{
+    //   //jwt cookie with session and can add people to session as well
+    //   res.send()
+    // })
+
+  }
```
Comment (yfcheng1) [inline server/controllers/orderSession/index.js:102]: What is this session code for?

File: server/index.js [hunk 10]
```diff
@@ -5,12 +5,13 @@ const cookieParser = require('cookie-parser');
...
 const app = express();
 var port = 3001;
-
+var cors = require('cors');
 app.use(express.json());
 app.use(cookieParser());
 app.use(express.urlencoded({ extended: true }));
 app.use(express.static(path.join(__dirname, "../client/build")));
 
+app.use(cors())
```
Comment (yfcheng1) [inline server/index.js:14]: Thanks for implementing cors for us!

**Review by yfcheng1 — CHANGES_REQUESTED** [review]
Great job Grant! I left couple comments on:
1. styles
2. console.log(s)
3. In session route, the user structure seems incorrect
4. there is one session.find code in the route needs checking
5. Do we need the npmignore file?
6. Thanks for implement cors for us

Remember to merge to main again before you push.

**Comment by yfcheng1** [comment]
Grant, let me close your pull request and we can merge with Tan's tomorrow! Thanks for the work!

**Pull request closed without merging**